core/domain/state_domain.py

```diff
@@ -1310,15 +1341,32 @@ def _update_content_ids_in_assets(self, old_ids_list, new_ids_list):
         for content_id in content_ids_to_delete:
             if not content_id in self.content_ids_to_audio_translations:
                 raise Exception(
-                    'The content_id %s does not exist.' % content_id)
+                    'The content_id %s does not exist in '
+                    'content_ids_to_audio_translations.' % content_id)
+            elif not content_id in (
+                    self.content_translations.get_all_content_ids_for_translation()): # pylint: disable=line-too-long
```

Comment:
If I'm reading this right, self.content_translations has an internal property called self.content_translations. This seems like a recipe for confusion and I think some renaming is needed.

Also, let's figure out a way to not have the pylint pragma, here and below.